packages/palette/src/elements/Modal/Modal.tsx

```diff
@@ -92,13 +104,11 @@ export const Modal: SFC<ModalProps> = ({
 
   // Listens for props to change and fades in new content
   useEffect(() => {
-    if (previousChildren && children !== previousChildren) {
-      setSpringContentAnimation({
-        opacity: 0,
-        onRest: () => updateVisibleContent(),
-      })
-    }
-  }, [children])
+    setSpringContentAnimation({
+      opacity: 0,
+      onRest: () => updateVisibleContent(),
+    })
+  }, [refreshModalContent])
```

Comment:
This feels kinda weird to me, like it might confuse devs further down the road who are wondering why things aren't updating right. I understand the reasoning (pass this vs checking diffs against specific content props) but it's a little less intuitive.

Reply:
Yeah ideally we'd check vs `children` but because the `children` here are so variable it's hard to pinpoint an exact prop to check against. If we check `children` vs `previousChildren` without a specific key then then it's always returned true.

Reply:
Yah, `children` being a complex object / function it's not comparable (only shallow comparisons are possible). 🤔Yeah, I don't have a good answer here.